runtime-k3d/src/runtime/controlplane.py

```diff
@@ -0,0 +1,280 @@
+# Copyright (c) 2023 Robert Bosch GmbH
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import subprocess
+from typing import List
+
+from yaspin.core import Yaspin
+
+from velocitas_lib import get_package_path, get_script_path, require_env
+
+
+def registry_exists() -> bool:
+    """Check if the K3D registry exists.
+
+    Returns:
+        bool: True if the registry exists, False if not.
+    """
+    return (
+        subprocess.call(
+            ["k3d", "registry", "get", "k3d-registry.localhost"],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+        == 0
+    )
+
+
+def create_registry():
+    """Create the K3D registry."""
+    subprocess.check_call(
+        ["k3d", "registry", "create", "registry.localhost", "--port", "12345"],
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+    )
+
+
+def delete_registry():
+    """Delete the K3D registry."""
+    subprocess.check_call(
+        ["k3d", "registry", "delete", "k3d-registry.localhost"],
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+    )
+
+
+def cluster_exists() -> bool:
+    """Check if the K3D cluster exists.
+
+    Returns:
+        bool: True if the cluster exists, False if not.
+    """
+    return (
+        subprocess.call(
+            ["k3d", "cluster", "get", "cluster"],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+        == 0
+    )
+
+
+def create_cluster(config_dir_path: str):
+    """Create the cluster with the given config dir.
+
+    Args:
+        config_dir_path (str): The path to the config directory.
+    """
+    has_proxy: bool = os.getenv("HTTP_PROXY") is not None
+
+    extra_proxy_args: List[str] = list()
+    if has_proxy:
+        print("Creating cluster with proxy configuration.")
+        http_proxy = os.getenv("HTTP_PROXY")
+        https_proxy = os.getenv("HTTPS_PROXY")
+        no_proxy = os.getenv("NO_PROXY")
+        extra_proxy_args = [
+            "-e",
+            f"HTTP_PROXY={http_proxy}@server:0",
+            "-e",
+            f"HTTPS_PROXY={https_proxy}@server:0",
+            "-e",
+            f"NO_PROXY={no_proxy}@server:0",
+        ]
+    else:
+        print("Creating cluster without proxy configuration.")
```

Comment:
Could be simplified like suggested in [runtime-k3d/src/deployment/build_vehicleapp.py](https://github.com/eclipse-velocitas/devenv-runtimes/pull/10/files#diff-44bfc04e16473edc68bfe5511540e799be9aeec980acd15537965b8c1d3bc1ed)